src/common/transformations/src/transformations/common_optimizations/concat_reduce_fusion.cpp

```diff
@@ -0,0 +1,155 @@
+
+    auto squeeze_axes_pattern = pattern::wrap_type<opset8::Constant>();
+    auto squeeze_pattern = pattern::wrap_type<opset8::Squeeze>({eltwise_pattern, squeeze_axes_pattern});
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto & pattern_map = m.get_pattern_map();
+        auto eltwise = pattern_map.at(eltwise_pattern);
+
+        if (!eltwise) {
+            return false;
+        }
+
+        size_t eltwise_inputs_size = eltwise->get_input_size();
+        for (size_t input_index = 0; input_index < eltwise_inputs_size; ++input_index) {
+            auto input_node = eltwise->get_input_node_shared_ptr(input_index);
+            if (!is_type<opset8::Constant>(input_node) && !is_type<opset8::Unsqueeze>(input_node)) {
+                return false;
+            }
+        }
+
+        auto squeeze = pattern_map.at(squeeze_pattern);
+        if (!squeeze) {
+            return false;
+        }
+
+        ngraph::OutputVector eltwise_inputs;
+        for (size_t input_index = 0; input_index < eltwise_inputs_size; ++input_index) {
+            auto eltwise_input = eltwise->input_value(input_index);
+            std::shared_ptr<Node> new_input_node = squeeze->clone_with_new_inputs({eltwise_input, squeeze->input_value(1)});
+            ngraph::copy_runtime_info(squeeze, new_input_node);
+            eltwise_inputs.push_back(new_input_node);
+        }
+
+        auto new_eltwise = eltwise->copy_with_new_inputs(eltwise_inputs);
+        new_eltwise->set_friendly_name(eltwise->get_friendly_name());
+        ngraph::copy_runtime_info({eltwise, squeeze}, new_eltwise);
+        ngraph::replace_node(squeeze, new_eltwise);
+        return true;
+    };
+
+    auto m = std::make_shared<pattern::Matcher>(squeeze_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+ngraph::pass::ConcatReduceFusionWithoutFolding::ConcatReduceFusionWithoutFolding() {
+    MATCHER_SCOPE(ConcatReduceFusionWithoutFolding);
+
+    auto concat_pattern = ngraph::pattern::wrap_type<opset8::Concat>({pattern::any_input(), pattern::any_input()});
+    auto reduce_axes_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    auto reduce_pattern = ngraph::pattern::wrap_type<opset8::ReduceMin, opset8::ReduceMax>({concat_pattern, reduce_axes_pattern});
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto concat = as_type_ptr<opset8::Concat>(pattern_map.at(concat_pattern).get_node_shared_ptr());
+
+        ReduceType reduce_type = get_reduce_type(pattern_map.at(reduce_pattern).get_node_shared_ptr());
+        if (reduce_type == ReduceType::NONE) {
+            return false;
+        }
+
+        auto reduce = as_type_ptr<op::util::ArithmeticReductionKeepDims>(pattern_map.at(reduce_pattern).get_node_shared_ptr());
+
+        if (!reduce || !concat) {
+             return false;
+        }
```

Comment:
```suggestion
        auto concat = as_type_ptr<opset8::Concat>(pattern_map.at(concat_pattern).get_node_shared_ptr());
        auto reduce = as_type_ptr<op::util::ArithmeticReductionKeepDims>(pattern_map.at(reduce_pattern).get_node_shared_ptr());
        if (!reduce || !concat) return false;
        
        ReduceType reduce_type = get_reduce_type(reduce);
        if (reduce_type == ReduceType::NONE) return false;
```

Reply:
BTW if we have `switch (reduce_type) {` we don't have to check type here:
```cpp
        ReduceType reduce_type = get_reduce_type(reduce);
        if (reduce_type == ReduceType::NONE) return false;
```

Reply:
Corrected, thanks